tests/assets/smb_example_config.yml

```diff
@@ -0,0 +1,26 @@
+root-dir: ~/Documents/HSR/semester_06
+
+global-ignore:
+  - Thumbs.db
+  - .DS_Store
+
+plugins:
```

Comment:
`plugins` is a bad name, and it shows in the code - the difference between a `plugin`, a `plugin_name`, a `plugin_type`, a `plugin_usage`, etc. etc. is way too subtle and confusing.

Here's what I'm thinking:

A *plugin* is a backend/handler for an "external service". `smb` is one. `moodle` is one. Whatever *Leonie* writes is one.

Thus, this shouldn't be named `plugin`. The writer of the config file doesn't define or write any plugins.

What about `connections:` instead? I feel like that'd make the code much readable as well, and as an user it makes sense to define a "connection".